aidermacs.el

```diff
@@ -667,12 +670,36 @@ Returns a deduplicated list of such file names."
 ;;;###autoload
 (defun aidermacs-list-added-files ()
   "List all files currently added to the chat session.
-Sends the \"/ls\" command and returns the list of files via callback."
+Sends the \"/ls\" command and displays the results in a Dired buffer."
   (interactive)
   (aidermacs--get-files-in-session
    (lambda (files)
-     (message "%s" (prin1-to-string files))
-     files)))
+     (let ((buf-name (aidermacs-get-buffer-name nil " Files")))
+       ;; Unfortunately find-dired-with-command doesn't allow us to specify the
+       ;; buffer name, so we manually rename it after the fact and recreate it
+       ;; on each call.
+       (when (get-buffer buf-name)
+         (kill-buffer buf-name))
+       (if files
+           (let* ((git-root (vc-git-root default-directory))
+                  (files-arg (mapconcat #'shell-quote-argument files " "))
+                  (cmd (format "find %s %s" files-arg (car find-ls-option))))
+             (find-dired-with-command git-root cmd)
+             (let ((buf (get-buffer "*Find*")))
+               (when buf
+                 (with-current-buffer buf
+                   (rename-buffer buf-name)
+                   (save-excursion
+                     ;; The executed command is on the 2nd line; it can get
+                     ;; quite long, so we delete it to avoid cluttering the
+                     ;; buffer.
+                     (goto-line 2)
+                     (when (looking-at "^ *find " t)
+                       (let ((inhibit-read-only t))
+                         (delete-region (line-beginning-position) (line-end-position)))))
+                   (setq revert-buffer-function
+                         (lambda (&rest _) (aidermacs-list-added-files)))))))
+         (message "No files added to the chat session"))))))
```

Comment:
The hackiness of this (managing the buffer name, deleting the `find` invocation) could be reduced by having a custom implementation of `find-dired-with-command`, but that looked quite cumbersome so I thought this was a reasonable first try.